### ISSUE TITLE
asyncio.gather missing

### ISSUE THREAD

**Issue by toejough** [python/typeshed#453]
When running mypy against a function containing `gathered = asyncio.gather(*tasks)`, an error is printed: `error: "module" has no attribute "gather"`

The cpython docs describe the `gather` function [here](https://docs.python.org/3/library/asyncio-task.html#asyncio.gather), and if run under python 3.5.2, the function is accessible: 

```python
>>> import asyncio
>>> type(asyncio.gather)
<class 'function'>
>>>
```

Searching in typeshed for "gather", however, turns up no results.

**Comment by gvanrossum** [comment]
Thanks for the report -- this definitely looks like an oversight. Do you feel up to submitting a PR to right this?

**Comment by toejough** [comment]
Sure!  Though I'm just getting started with mypy, so the first draft of the PR might be a bit rough.  I'll make some time to submit something this week.

**Comment by claws** [comment]
I think that to resolve this issue the ``typeshed/stdlib/3.4/asyncio/tasks.pyi`` file would be updated with:
```
def gather(*coros_or_futures: Sequence[Union[Future[_T], Generator[Any, None, _T]]], loop: AbstractEventLoop = ..., return_exceptions: bool = False) -> Future[_T]: ...
```

While I was looking at how this issue could be resolved I noticed other items are missing from the ``task.pyi`` file. I think the items below cover most of these and would help to improve the asyncio typeshed coverage.
```
def as_completed(fs: Sequence[Future[_T]], *, loop: AbstractEventLoop = ..., timeout=None) -> Iterator[Generator[Any, None, _T]]:
if sys.version_info >= (3, 5):
    def ensure_future(coro_or_future: Union[Future[_T], Generator[Any, None, _T]], *, loop: AbstractEventLoop = ...) -> Future[_T]: ...
def gather(*coros_or_futures: Sequence[Union[Future[_T], Generator[Any, None, _T]]], loop: AbstractEventLoop = ..., return_exceptions: bool = False) -> Future[_T]: ...
def run_coroutine_threadsafe(coro: Generator[Any, None, _T], loop: AbstractEventLoop) -> Future[_T]: ...
def shield(arg: Union[Future[_T], Generator[Any, None, _T]], *, loop: AbstractEventLoop = ...) -> Future[_T]: ...

class _GatheringFuture(Future[_T], Generic[_T]):
    def __init__(self, children: Sequence[Union[Future[_T], Generator[Any, None, _T]]], *, loop: AbstractEventLoop = ...):
    def cancel(self) -> bool:
```

How to proceed? Just create a PR containing ``gather`` and raise a separate issue for the others? Or create a PR containing all of them?

**Comment by gvanrossum** [comment]
You can do it all in one PR.

**Comment by gvanrossum** [comment]
Fixed by #511. (Note: if you include "Fixes $453" in the body of the commit message, the bug will be automatically closed when the PR is merged.)
